src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs

```diff
@@ -656,37 +663,62 @@ private void OnTreeNodeGUI(BranchTreeNode node)
 
                 if (Event.current.clickCount > 1 && mode == BranchesMode.Default)
                 {
-                    if (node.Type == NodeType.LocalBranch &&
-                        EditorUtility.DisplayDialog(ConfirmSwitchTitle, String.Format(ConfirmSwitchMessage, node.Name), ConfirmSwitchOK,
-                            ConfirmSwitchCancel))
+                    if (node.Type == NodeType.LocalBranch)
                     {
-                        GitClient.SwitchBranch(node.Name)
-                            .FinallyInUI((success, e) =>
-                            {
-                                if (success)
-                                    Refresh();
-                                else
+                        if (EditorUtility.DisplayDialog(ConfirmSwitchTitle, String.Format(ConfirmSwitchMessage, node.Name), ConfirmSwitchOK, ConfirmSwitchCancel))
+                        {
+                            GitClient.SwitchBranch(node.Name)
+                                .FinallyInUI((success, e) =>
                                 {
-                                    EditorUtility.DisplayDialog(Localization.SwitchBranchTitle,
-                                        String.Format(Localization.SwitchBranchFailedDescription, node.Name),
-                                    Localization.Ok);
-                                }
-                            }).Start();
+                                    if (success)
+                                    {
+                                        Refresh();
+                                    }
+                                    else
+                                    {
+                                        EditorUtility.DisplayDialog(Localization.SwitchBranchTitle,
+                                            String.Format(Localization.SwitchBranchFailedDescription, node.Name),
+                                            Localization.Ok);
+                                    }
+                                }).Start();
+                        }
                     }
                     else if (node.Type == NodeType.RemoteBranch)
                     {
-                        GitClient.CreateBranch(selectedNode.Name.Substring(selectedNode.Name.IndexOf('/') + 1), selectedNode.Name)
-                            .FinallyInUI((success, e) =>
+                        var indexOfFirstSlash = selectedNode.Name.IndexOf('/');
+                        var originName = selectedNode.Name.Substring(0, indexOfFirstSlash);
+                        var branchName = selectedNode.Name.Substring(indexOfFirstSlash + 1);
+
+                        if (Repository.LocalBranches.Any(localBranch => localBranch.Name == branchName))
```

Comment:
What we should be doing is checking if there's a branch that is tracking the same commit to determine whether the local branch exists, and offer to check that one out if it does. Probably something to consider for the future.

Reply:
Yea, I thought about that, I'll open up a future issue for that.